src/LuaTransformer.ts

```diff
@@ -4477,7 +4477,8 @@ export class LuaTransformer {
         }
 
         if (decorators.has(DecoratorKind.LuaTable)) {
-            return this.transformLuaTableProperty(expression);
+            const [luaTable] = this.parseLuaTableExpression(expression);
```

Comment:
That's not what I meant. `table.get` shouldn't be transformed just to `table`, because these aren't interchangeable in isolation. `transformLuaTableProperty` should allow only `length` property.
What I meant is that `getLuaTablePropertyName` can be renamed to `parseLuaTableExpression` and would split lua table reference and it's method name.

Reply:
Thanks for the clarification, looks like it neatened up nicely